### ISSUE TITLE
User can share their finished artboard on Tumblr

### ISSUE THREAD

**Issue by erasmus** [degeist/book-cover-generator#4]
Can be done with open graph + Tumblr's share button.

https://www.tumblr.com/docs/en/share_button

**Complex API integration alternative**
Tumblr account: 
Degeist (tumblr@degeist.com)

Register app:
https://www.tumblr.com/oauth/register

**Comment by erasmus** [comment]
Tumblr may need the abs path to the complete Cloudinary img before it's photo sharing will work

**Comment by erasmus** [comment]
@avorio Tumblr's share button sucks. It wrongly parses the cloudinary image, because it has multiple `/` in it. 

E.g. go to the review deployment of my experiment branch:
https://danandphil-pr-59.herokuapp.com/artboard/v1469650162/pl6dwyv7behgpmmgtvdd.jpg/?cutout=dp-cutout5.png&text=go%20outside

and click the second Tumblr link. The following is a little bit of the console log from the popup window opened by Tumblr's API:

<img width="957" alt="screen shot 2016-07-27 at 22 17 55" src="https://cloud.githubusercontent.com/assets/1915920/17190979/420b2590-5448-11e6-8d87-11dcb0996bd5.png">

Basically, it misinterprets the slash as a path indicator... I think we may need an absolutely named file for sharing on Tumblr. (note, that it's possible to upload a full Cloudinary URL, with tonnes of slashes, via the normal photo upload on Tumblr.com).

Any good ideas of how to (easy)fix?

**Comment by erasmus** [comment]
Shoot, it looks like it's an issue with the Tumblr API. I can't deal with the comma in the Cloudinary image URL.

Alternative is to apply the Cloudinary transformation on upload:
http://cloudinary.com/documentation/upload_images#eager_asynchronous_transformations

~~How much effort would it be to implement this upload method? @avorio~~

**Update:**

It's not so much applying the transformation on upload, but rather applying when final artboard is created. So not via the URL.

**Comment by avorio** [comment]
I’m afraid I’d need to change the backend to store each artboard as an object in the database that we could later on retrieve with a simpler [and Tumblr-friendly] URL, e.g. `/artboard/123.jpg`.

As per our Skype conversation this morning, we decided to park this issue for now and revisit it once the client has agreed to the extra steps necessary to get it working.

**Comment by erasmus** [comment]
Tumblr fixed their API. Closing.
